gulpfile.js

```diff
@@ -163,6 +163,31 @@ function build()
                 }
             });
             
+            callback(null, file);
+        }))
+        .pipe(through.obj(function(file, enc, callback)
+        {
+            // add defensive check to prevent Mojito gets loading multi times
+            if (!config.allowMultiInstance) {
+                let content = fs.readFileSync(`dist/assets/js/${containerName}.js`, 'utf8');
```

Comment:
Cheers @allmywant - this is a cool way of dealing with the issue! 

As there is some repetition in this block, would it be possible to refactor the code and wrap the guts of it in another function? E.g.
```
function insertDefensiveCheck(filename) {
    let content = fs.readFileSync(filename, 'utf8');
    let position = content.indexOf('Mojito =');
    if (position === -1) {
        position = content.indexOf('Mojito=');
    }
    if (position !== -1) {
        content = content.substr(0, position) + 'if (!window.Mojito || !Mojito.testObjects || !Object.keys(Mojito.testObjects).length) {' + content.substring(position) + '}';
        fs.writeFileSync(filename, content);
    }
}
```

```
// Inside the build function, replace the repetitive code block with:
if (!config.allowMultiInstance) {
    insertDefensiveCheck(`dist/assets/js/${containerName}.js`);
    insertDefensiveCheck(`dist/assets/js/${containerName}.pretty.js`);
}
```

Reply:
@dapperdrop 
Good idea, I've wrapped the code into a `insertPreventDuplicateLoadingCheck` function.